### PR TITLE
add mika

### PR DESCRIPTION
**Pull request by imring**
- [x] Talents
- [x] Passive talents
- [x] Constellation
- [ ] Frames/Hitlags/Hitboxes
- [ ] TODO's

### CODE REVIEW

File: internal/characters/mika/charge.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		CanBeDefenseHalted: true,
+		Mult:               charge[c.TalentLvlAttack()],
+	}
+
+	c.Core.QueueAttack(
+		ai,
+		combat.NewBoxHitOnTarget(c.Core.Combat.Player(), geometry.Point{Y: -0.1}, 2.8, 4.8),
+		chargeHitmark,
+		chargeHitmark,
+	)
```
Comment (unleashurgeek) [inline internal/characters/mika/charge.go:47]: make hitbox same as hu tao's CA. 0.8 radius circle on primary target

File: internal/characters/mika/charge.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+func (c *char) ChargeAttack(p map[string]int) action.ActionInfo {
+	ai := combat.AttackInfo{
+		ActorIndex:         c.Index,
+		Abil:               "Charge Attack",
+		AttackTag:          attacks.AttackTagExtra,
+		ICDTag:             attacks.ICDTagExtraAttack,
+		ICDGroup:           attacks.ICDGroupDefault,
+		StrikeType:         attacks.StrikeTypeSlash,
```
Comment (unleashurgeek) [inline internal/characters/mika/charge.go:33]:
```
ICDGroup: attacks.ICDGroupPoleExtraAttack,
StrikeType: attacks.StrikeTypeSpear,
```

File: internal/characters/mika/attack.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+func (c *char) Attack(p map[string]int) action.ActionInfo {
+	for i, mult := range attack[c.NormalCounter] {
+		ai := combat.AttackInfo{
+			ActorIndex:         c.Index,
+			Abil:               fmt.Sprintf("Normal %v", c.NormalCounter),
+			AttackTag:          attacks.AttackTagNormal,
+			ICDTag:             attacks.ICDTagNormalAttack,
+			ICDGroup:           attacks.ICDGroupDefault,
+			StrikeType:         attacks.StrikeTypeSlash,
```
Comment (unleashurgeek) [inline internal/characters/mika/attack.go:57]: * Attacks 3 and 4 are `StrikeTypeSpear`
* Attacks 1, 2, 5 are `StrikeTypeSlash`

File: internal/characters/mika/attack.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+// TODO: update frames, hitlags & hitboxes
+var (
+	attackFrames   [][]int
+	attackHitmarks = [][]int{{14}, {9}, {14}, {14, 27}, {34}}
+	// same between polearm and burst attacks so just use these arrays for both
+	attackHitlagHaltFrame = [][]float64{{0.02}, {0.02}, {0.02}, {0, 0}, {0.04}}
+	attackDefHalt         = [][]bool{{false}, {true}, {false}, {true, true}, {true}}
+	attackHitboxes        = [][]float64{{1.8, 2.8}, {2.5}, {1.8, 4}, {2.8}, {3}}
+	attackOffsets         = [][][]float64{{{0, -0.2}}, {{0, 0.5}}, {{0, -1.2}}, {{-0.5, 0.7}, {0.3, 0.7}}, {{0, 0}}}
+	attackFanAngles       = []float64{360, 270, 360, 360, 360}
```
Comment (unleashurgeek) [inline internal/characters/mika/attack.go:26]: updated based on dm. The `Attack` logic will need to be changed to use circle vs box when relevant (remove raiden burst comment). Notes on the NA attacks:
* attack 4 is a 2 hit attack with identical hitboxes for both
* attack 2 and 4 are circle hitboxes, the rest are boxes (boxes have width and height in `attackHitboxes`)
* only attack 4 has an angle (attack 2 is a 360)
```golang
attackHitlagHaltFrame = [][]float64{{0.02}, {0.02}, {0.02}, {0, 0}, {0.04}}
attackDefHalt         = [][]bool{{false}, {true}, {false}, {true, true}, {true}}
attackHitboxes        = [][][]float64{{{1.8, 2.5}}, {{1.6}}, {{2.5, 2.5}}, {{4}, {4}}, {{1.5, 5}}}
attackOffsets         = []float64{0, 0.2, 0.5, -0.3, 1.0}
attackFanAngles       = []float64{360, 360, 360, 45, 360}
```

**Review by unleashurgeek — COMMENTED** [review]
Make sure to also update characters on UI: https://github.com/genshinsim/gcsim/blob/main/app/src/Components/Character/characters.tsx

File: internal/characters/mika/asc.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+// Per the following circumstances, the Soulwind state caused by Starfrost Swirl will grant characters the Detector effect,
+// increasing their Physical DMG by 10% when they are on the field.
+// - If the Flowfrost Arrow hits more than one opponent, each additional opponent hit will generate 1 Detector stack.
+// - When a Rimestar Shard hits an opponent, it will generate 1 Detector stack. Each Rimestar Shard can trigger the effect 1 time.
+//
+// The Soulwind state can have a maximum of 3 Detector stacks, and if Starfrost Swirl is cast again during this duration, the pre-existing
+// Soulwind state and all its Detector stacks will be cleared.
+func (c *char) a1() {
+	m := make([]float64, attributes.EndStatType)
+	for _, char := range c.Core.Player.Chars() {
```
[the remark this answers is not in the document]
Comment (imring) [reply internal/characters/mika/asc.go:38]: i don't think so
> will grant characters the Detector effect

**Review by unleashurgeek — APPROVED** [review]

**Pull request merged**